src/benchmark_runner.cc

```diff
@@ -140,23 +141,81 @@ void RunInThread(const BenchmarkInstance* b, IterationCount iters,
   manager->NotifyThreadComplete();
 }
 
+std::variant<int, double> ParseBenchMinTime(const std::string& value) {
+  if (value.empty()) return 0;
+
+  if (value.back() == 'x') {
+    std::string num_iters_str = value.substr(0, value.length() - 1);
+    int num_iters = std::atoi(num_iters_str);
+
+    // std::atoi doesn't provide useful error messages, so we do some
+    // sanity checks.
+    // Also check that the iters is not negative.
+    BM_CHECK(num_iters > 0 && !(num_iters == 0 && num_iters_str != "0"))
+        << "Malformed iters value passed to --benchmark_min_time: `" << value
+        << "`. Expected --benchmark_min_time=<integer>x.";
+    return num_iters;
+  }
+
+  std::string min_time_str;
+  if (value.back() != 's') {
+    BM_VLOG(2) << "Value passed to --benchmark_min_time should have a suffix. "
+                  "Eg., `30s` for 30-seconds.";
+    min_time_str = value;
+  } else {
+    min_time_str = value.substr(0, value.length() - 1);
+  }
+  double min_time = std::atof(min_time_str);
+
+  BM_CHECK(min_time > 0 &&
+           !(min_time == 0 && (min_time_str != "0" || min_time_str != "0.0")))
+      << "Malformed seconds value passed to --benchmark_min_time: `" << value
+      << "`. Expected --benchmark_min_time=<float>x.";
+  return min_time;
+}
+
+double GetMinTime(const benchmark::internal::BenchmarkInstance& b) {
+  if (!IsZero(b.min_time())) return b.min_time();
+
+  std::variant<int, double> iters_or_time =
+      ParseBenchMinTime(FLAGS_benchmark_min_time);
+
+  // If the flag was used to specify number of iters, then return 0 for time.
+  if (iters_or_time.index() == 0) return 0;
```

Comment:
this check is wrong. someone could pass in "0x" expecting it to be a minimum of 0 iterations.

Reply:
I'm not sure I understand why it's wrong.

If someone passes "<anything>x", then we interpret the number as the number of iterations.
Whether it's zero or not doesn't make a difference. (ie., doesn't change the fact that they are specifying the number of iterations and not the time limit). 

This check is just checking to see whether the value of the flag is for iterations or time. (`.index() == 0` is to check which member of the variant is active.  0 is for the first member `int` and `1` is for second member `double`)